hedera-node/src/main/java/com/hedera/services/state/logic/TopLevelTransition.java

```diff
@@ -73,7 +74,7 @@ public TopLevelTransition(
 
 	@Override
 	public void run() {
-		final var accessor = txnCtx.accessor();
+		final var accessor = (PlatformTxnAccessor) txnCtx.accessor();
```

Comment:
There's definitely work to do with the whole `TransactionContext` API, but here also it feels safer to have two different methods to get the triggered vs. top-level accessors...so here `txnCtx.platformAccessor()` that returns `PlatformTxnAccessor`; and in `TriggeredTransition`, a `txnCtx.accessor()` or something.

Definitely work to do... 🤔